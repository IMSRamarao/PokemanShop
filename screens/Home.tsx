import {View, Text, ScrollView, StyleSheet} from 'react-native';
import React, {useEffect, useState} from 'react';
import Header from '../src/components/Header';
import Products from '../src/components/Products';

interface Products {
  id: number;
  title: string;
  price: number;
  description: string;
  category: string;
  image: string;
  rating: number;
}
[];

const Home = () => {
  const [product, setProduct] = useState<Products[]>([]);
  const [loading, setLoading] = useState(true);
  const fetchData = async () => {
    const resp = await fetch(
      'https://fakestoreapiserver.reactbd.com/amazonproducts',
    );
    const data = await resp.json();
    setProduct(data);
    setLoading(false);
  };
  useEffect(() => {
    fetchData();
  }, []);
  return (
    <View >
      {loading && (
        <View style={styles.loader}>
          <Text style={styles.loadingData}>Data is loading...</Text>
        </View>
      )}

      <ScrollView>
        <View style={styles.titleContainer}>
          <Text style={styles.title}>Pokeman Store</Text>
          <Text style={styles.app_Des}>
            This is Pokeman app to to get the data of pokeman
          </Text>
        </View>
        <View style={styles.list}>
          {product.map(item => (
            <Products key={item.id} item={item} />
          ))}
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  loader: {
    height: '70%',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  titleContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    padding: 10,
  },
  app_Des: {
    fontSize: 16,
    fontWeight: '400',
    padding: 10,
  },
  list: {
    width: '100%',
  },
  loadingData: {
    fontSize: 25,
    fontWeight: 'bold',
  },
});

export default Home;
