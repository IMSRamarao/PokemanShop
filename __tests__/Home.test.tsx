import 'react-native';
import React from 'react';
import Home from '../screens/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.useFakeTimers();
test('Home check state data', () => {
  let homeData = renderer.create(<Home />).toJSON();
  expect(homeData.children).toHaveLength(2);
});

test('Home snapshots', () => {
  const snap = renderer.create(<Home />).toJSON();
  expect(snap).toMatchSnapshot();
});
