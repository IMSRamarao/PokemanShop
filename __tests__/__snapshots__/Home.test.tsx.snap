// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshots 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "height": "70%",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Data is loading...
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
          Pokeman Store
        </Text>
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "400",
              "padding": 10,
            }
          }
        >
          This is Pokeman app to to get the data of pokeman
        </Text>
      </View>
      <View
        style={
          {
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
